### PR TITLE
Add zookeeper-testing module

### PR DESCRIPTION
**Pull request by ryanhall07**
and various changes to suppport using misk-zookeeper outside of misk

### CODE REVIEW

File: misk-zookeeper-testing/build.gradle [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    classpath dep.kotlinNoArgPlugin
+  }
+}
+
+apply plugin: 'kotlin-jpa'
+
+dependencies {
+  compile dep.curatorFramework
+  compile dep.zookeeper
+  compile  dep.docker
```
Comment (wesleyk) [inline misk-zookeeper-testing/build.gradle]: /nit extra space

File: misk-zookeeper-testing/src/main/kotlin/misk/zookeeper/testing/ZkTestModule.kt [hunk 2]
```diff
@@ -57,7 +59,8 @@ class ZkTestModule(
     }
 
     companion object {
-      private val sharedZookeeper = EmbeddedZookeeper(zkPortKey)
+      private val sharedZookeeper = EmbeddedZookeeper(
+          zkPortKey)
```
Comment (wesleyk) [inline misk-zookeeper-testing/src/main/kotlin/misk/zookeeper/testing/ZkTestModule.kt]: /nit one line

File: misk-zookeeper-testing/build.gradle (hunk 1 above)
Comment (ryanhall07) [reply misk-zookeeper-testing/build.gradle]: weird. intellij did this during the refactor. bad intellij.

File: misk-zookeeper-testing/src/main/kotlin/misk/zookeeper/testing/ZkTestModule.kt (hunk 2 above)
Comment (ryanhall07) [reply misk-zookeeper-testing/src/main/kotlin/misk/zookeeper/testing/ZkTestModule.kt]: weird. intellij did this during the refactor. bad intellij.

**Pull request merged**